addons/misra.py

```diff
@@ -850,6 +850,17 @@ def isNoReturnScope(tok):
     return False
 
 
+# Return the token which the value is assigned to
+def getAssignedVariableToken(valueToken):
+    if not valueToken:
+        return None
+    if not valueToken.astParent:
+        return None
+    operator = valueToken.astParent
+    if not operator.isAssignmentOp:
```

Comment:
I think this would not ensure warnings for such code:

    char *str = 1 + "hello world";

Reply:
Ok, is this a valid test case? If so I will extend the function which finds the variable being assigned.

Reply:
Fixed in latest commit.